### ISSUE TITLE
AddMult bilinearform support for PA,EA, etc... 

### ISSUE THREAD

**Issue by artv3** [mfem/mfem#2870]
Hi all, 

It would be nice to have general support for AddMult for the different assembly types in bilinearform:

https://github.com/mfem/mfem/blob/baf9090c887e08058f0e93c01274b4cb4e069a19/fem/bilinearform.hpp#L273-L275

**Comment by artv3** [comment]
This would still be good to do.

**Comment by victor-decaria-nnl** [comment]
Can this issue be resolved even if the solution is to just do something like this in the short term?
```c++
void BilinearForm::AddMult(const Vector &x, Vector &y, const double a) const
{
   if (ext)
   {
      ext->AddMult(x, y, a); // Use default Operator::AddMult.
   }
   else
   {
      mat->AddMult(x, y, a);
   }
}
 ```
 As it stands, this is kind of a bug since `mat` doesn't get assembled if an extension is used, and my workaround is to call `Operator::AddMult` everywhere to avoid this.

**Comment by v-dobrev** [comment]
@victor-decaria-nnl, yes, we can do that. We should check to make sure the behavior is consistent between the the two case in the `if` statement -- for example when the Mesh is nonconforming.

Do you want to propose a PR?  Alternatively, we can try to resurrect @artv3's PR #3045 (I'm not sure why Arturo gave up on that and closed it).

**Comment by victor-decaria-nnl** [comment]
@v-dobrev #3045 looks a bit involved. I can propose a workaround PR for the time being.
